kotlinsample/src/main/java/com/airbnb/epoxy/kotlinsample/helpers/ViewBindingEpoxyModelWithHolder.kt

```diff
@@ -0,0 +1,51 @@
+package com.airbnb.epoxy.kotlinsample.helpers
+
+import android.view.View
+import androidx.viewbinding.ViewBinding
+import com.airbnb.epoxy.EpoxyHolder
+import com.airbnb.epoxy.EpoxyModelWithHolder
+import java.lang.reflect.Method
+import java.lang.reflect.ParameterizedType
+import java.util.concurrent.ConcurrentHashMap
+
+abstract class ViewBindingEpoxyModelWithHolder<in T : ViewBinding> : EpoxyModelWithHolder<ViewBindingHolder>() {
+    // Using reflection to get the static binding method.
+    // Lazy so it's computed only once by instance, when the 1st ViewHolder is actually created.
+    private val actualTypeOfThis by lazy { (this::class.java.genericSuperclass as Class<*>).genericSuperclass as ParameterizedType }
+    private val kClass by lazy { (actualTypeOfThis.actualTypeArguments[0] as Class<ViewBinding>) }
+    private val bindingMethod by lazy { getBindMethod(kClass) }
+
+    @Suppress("UNCHECKED_CAST")
+    override fun bind(holder: ViewBindingHolder) {
+        (holder.viewBinding as T).bind()
+    }
+
+    abstract fun T.bind()
+
+    override fun createNewHolder(): ViewBindingHolder {
+        return ViewBindingHolder(bindingMethod)
```

Comment:
I would do the binding method lookup in the viewholder instead of the model because:
1. many more models are created than viewholders. models are constantly rebuilt while viewholders are reused, which reduces the reflection
2. the method is only used in the viewholder, and the model doesn't need to know about it